server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -717,6 +718,7 @@ public Node start() throws NodeValidationException {
         // Start the transport service now so the publish address will be added to the local disco node in ClusterService
         TransportService transportService = injector.getInstance(TransportService.class);
         transportService.getTaskManager().setTaskResultsService(injector.getInstance(TaskResultsService.class));
+        transportService.getTaskManager().setTaskCancellationService(new TaskCancellationService(transportService));
```

Comment:
why isn't it instantiated when we create the TaskManager (TransportService#createTaskManager) ?

Reply:
I tried to avoid exposing TransportService when it's not fully initialized yet. Also, there are several circular dependencies between TransportService, RequestHandlerRegistry, TaskCancellationService, and TaskManager. I will look into how to pass TaskCancellationService to the constructor of TaskManager.